### PR TITLE
[Sprint 1] Events tab main screen

### PR DESCRIPTION
**Pull request by tranminhtri911**
https://dev.framgia.com/redmine/users/1671
Handle events tab main screen
Effect : None

### CODE REVIEW

File: app/src/main/java/com/framgia/forder/screen/main/MainActivity.java [hunk 1]
```diff
@@ -2,14 +2,18 @@
...
 import com.framgia.forder.databinding.ActivityMainBinding;
 import com.framgia.forder.screen.BaseActivity;
 
+import static com.framgia.forder.utils.utils.addFragmentToActivity;
+
 /**
  * Main Screen.
  */
-public class MainActivity extends BaseActivity {
+public class MainActivity extends BaseActivity implements View.OnClickListener {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/main/MainActivity.java]: remove ``implements View.OnClickListener``

File: app/src/main/java/com/framgia/forder/screen/main/MainActivity.java [hunk 2]
```diff
@@ -37,4 +41,31 @@ protected void onStop() {
         mViewModel.onStop();
         super.onStop();
     }
+
+    @Override
+    public void onClick(View v) {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/main/MainActivity.java]: move to ``MainViewModel``

File: app/src/main/java/com/framgia/forder/screen/main/MainActivity.java [hunk 3]
```diff
@@ -37,4 +41,31 @@ protected void onStop() {
...
+                break;
+            case R.id.layout_profile:
+                //TODO new Fragment Profile
+                break;
+            default:
+                break;
+        }
+    }
+
+    public void addFragment(Fragment fragment) {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/main/MainActivity.java]: move to ``MainViewModel``

File: app/src/main/java/com/framgia/forder/screen/main/MainActivity.java [hunk 4]
```diff
@@ -37,4 +41,31 @@ protected void onStop() {
...
+        addFragmentToActivity(getSupportFragmentManager(), fragment, R.id.layout_content);
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/main/MainActivity.java]: Please declare this method in ``Navigator.java`` and use ``Navigator`` to redirect screen

File: app/src/main/java/com/framgia/forder/utils/utils.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+import android.support.annotation.NonNull;
+import android.support.v4.app.Fragment;
+import android.support.v4.app.FragmentManager;
+
+/**
+ * Created by tri on 13/04/2017.
+ */
+
+public class utils {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/utils/utils.java]: remove this file

File: app/src/main/res/layout/layout_footer.xml [hunk 6]
```diff
@@ -20,6 +25,7 @@
             >
             <LinearLayout
                 android:id="@+id/layout_home"
+                android:onClick="onClick"
```
Comment (daolq3012) [inline app/src/main/res/layout/layout_footer.xml]: change to
```xml
             android:onClick="@{viewModel::onFooterClicked}"
```

File: app/src/main/res/layout/layout_footer.xml [hunk 7]
```diff
@@ -33,6 +39,7 @@
             </LinearLayout>
             <LinearLayout
                 android:id="@+id/layout_seach"
+                android:onClick="onClick"
```
Comment (daolq3012) [inline app/src/main/res/layout/layout_footer.xml]: change to 
```xml
             android:onClick="@{viewModel::onFooterClicked}"
```

File: app/src/main/res/layout/layout_footer.xml [hunk 8]
```diff
@@ -49,6 +56,7 @@
                 />
             <LinearLayout
                 android:id="@+id/layout_notify"
+                android:onClick="onClick"
```
Comment (daolq3012) [inline app/src/main/res/layout/layout_footer.xml]: change to 
```xml
             android:onClick="@{viewModel::onFooterClicked}"
```

File: app/src/main/res/layout/layout_footer.xml [hunk 9]
```diff
@@ -62,6 +70,7 @@
             </LinearLayout>
             <LinearLayout
                 android:id="@+id/layout_profile"
+                android:onClick="onClick"
```
Comment (daolq3012) [inline app/src/main/res/layout/layout_footer.xml]: change to 
```xml
             android:onClick="@{viewModel::onFooterClicked}"
```

File: app/src/main/res/layout/layout_footer.xml [hunk 10]
```diff
@@ -81,6 +90,7 @@
             android:layout_height="@dimen/dp_60"
             android:layout_centerInParent="true"
             android:background="@drawable/center_tab_footer"
+            android:onClick="onClick"
```
Comment (daolq3012) [inline app/src/main/res/layout/layout_footer.xml]: change to 
```xml
             android:onClick="@{viewModel::onFooterClicked}"
```

File: app/src/main/java/com/framgia/forder/screen/main/MainActivity.java [hunk 11]
```diff
@@ -17,7 +17,7 @@
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
-        mViewModel = new MainViewModel();
+        mViewModel = new MainViewModel(getSupportFragmentManager());
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/main/MainActivity.java]: remove this param, Please pass the instance of ``Navigator`` to ViewModel and using it to navigator screen

File: app/src/main/java/com/framgia/forder/screen/main/MainViewModel.java [hunk 12]
```diff
@@ -25,4 +36,30 @@ public void onStop() {
...
+                break;
+            case R.id.layout_profile:
+                //TODO new Fragment Profile
+                break;
+            default:
+                break;
+        }
+    }
+
+    public void addFragment(Fragment fragment) {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/main/MainViewModel.java]: remove this method

File: app/src/main/java/com/framgia/forder/utils/navigator/Navigator.java [hunk 13]
```diff
@@ -128,4 +129,9 @@ private void setFragmentTransactionAnimation(FragmentTransaction transaction,
         int START = 0x00;
         int FINISH = 0x01;
     }
+
+    public static void addFragmentToActivity(@NonNull FragmentManager fragmentManager,
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/utils/navigator/Navigator.java]: none static

File: app/src/main/java/com/framgia/forder/screen/main/MainActivity.java [hunk 14]
```diff
@@ -17,7 +17,7 @@
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
-        mViewModel = new MainViewModel();
+        mViewModel = new MainViewModel(new MainViewPagerAdapter(getSupportFragmentManager()));
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/main/MainActivity.java]: Please change to
```java
MainViewPagerAdapter adapter = new MainViewPagerAdapter(getSupportFragmentManager());
mViewModel = new MainViewModel(adapter);
```

File: app/src/main/res/layout/layout_footer.xml [hunk 15]
```diff
@@ -4,9 +4,14 @@
...
 
     <data>
+        <variable
+            name="viewModel"
+            type="com.framgia.forder.screen.main.MainViewModel"
+            />
     </data>
 
     <RelativeLayout
+        android:id="@+id/layout_footer"
```
Comment (daolq3012) [inline app/src/main/res/layout/layout_footer.xml]: remove this id, it unnecessary

File: app/src/main/res/layout/layout_footer.xml [hunk 16]
```diff
@@ -81,6 +91,8 @@
             android:layout_height="@dimen/dp_60"
             android:layout_centerInParent="true"
             android:background="@drawable/center_tab_footer"
+            android:onClick="@{viewModel::onFooterClicked}"
+            android:scaleType="center"
```
Comment (daolq3012) [inline app/src/main/res/layout/layout_footer.xml]: remove this line. Should be keep image with origin size

File: app/src/main/res/layout/layout_footer.xml (hunk 15 above)
Comment (tranminhtri911) [inline app/src/main/res/layout/layout_footer.xml]: cần id để em dùng bên layout main anh ơi

File: app/src/main/java/com/framgia/forder/screen/main/FragmentContainer.java [hunk 17]
```diff
@@ -0,0 +1,60 @@
...
+        fragment.setArguments(bundle);
+        return fragment;
+    }
+
+    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
+            @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_container, container, false);
+        Navigator navigator = new Navigator(this);
+
+        switch (getArguments().getInt(EXTRA_TAB_FOOTER)) {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/main/FragmentContainer.java]: change to
```java
@Tab int tab = getArguments().getInt(EXTRA_TAB_FOOTER);
switch (tab) {
...
}
```

File: app/src/main/java/com/framgia/forder/screen/login/LoginActivity.java [hunk 18]
```diff
@@ -17,7 +18,7 @@
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
-        mViewModel = new LoginViewModel();
+        mViewModel = new LoginViewModel(new Navigator(this));
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/login/LoginActivity.java]: change to
```java
Navigator navigator = new Navigator(this);
mViewModel = new LoginViewModel(navigator);
```

File: app/src/main/java/com/framgia/forder/screen/login/LoginViewModel.java [hunk 19]
```diff
@@ -25,4 +31,9 @@ public void onStop() {
     public void setPresenter(LoginContract.Presenter presenter) {
         mPresenter = presenter;
     }
+
+    public void onLoginButtonClicked(View view) {
+        //TODO change later
+        mNavigator.startActivity(MainActivity.class);
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/login/LoginViewModel.java]: add ``mNavigator.finish()`` in here

File: app/src/main/java/com/framgia/forder/screen/main/MainViewPagerAdapter.java [hunk 20]
```diff
@@ -0,0 +1,27 @@
...
+import android.support.v4.app.Fragment;
+import android.support.v4.app.FragmentManager;
+import android.support.v4.app.FragmentStatePagerAdapter;
+
+/**
+ * Created by tri on 13/04/2017.
+ */
+
+public class MainViewPagerAdapter extends FragmentStatePagerAdapter {
+    private final int TAB_NUMBER = 5;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/main/MainViewPagerAdapter.java]: ``constant`` must declare `` static final ...``
change to
```java
private static final int TAB_NUMBER = 5;
```

**Comment by daolq3012** [comment]
OK!

**Pull request merged**